fix: SMB remount issue

andyzhangx opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
3. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Porting PR(https://github.com/kubernetes/utils/pull/184) to csi-proxy.
There is a critical bug in original SMB mount logic:

   When SMB server password changed, there is no way for csi-proxy to remove original SMB server connection, only workaround is user log on to the Windows agent node, and run `Remove-SMBGlobalMapping` command manually to remove that connection. This PR could fix the issue.

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

**Special notes for your reviewer**:
/assign @jingxu97 

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
fix: SMB remount issue
```

andyzhangx commented:
@jingxu97 I think we should make this PR into v0.2.0 release

jingxu97 commented on internal/server/smb/server.go:
```diff
@@ -45,6 +45,22 @@ func (s *Server) NewSmbGlobalMapping(context context.Context, request *internal.
 		isMapped = false
 	}
 
+	valid, err := s.fsServer.PathValid(context, remotePath)
```
Another option is let the caller to perform this logic. csi-proxy just provide all APIs needed such as PathValid.

andyzhangx replied:
I think we could expose `PathValid` interface in filesystem, and for this issue, let's fix it first.

jingxu97 commented on internal/os/filesystem/api.go:
```diff
@@ -31,6 +33,21 @@ func (APIImplementor) PathExists(path string) (bool, error) {
+	return strings.HasPrefix(strings.ToLower(string(output)), "true"), nil
+}
+
+func (APIImplementor) PathValid(path string) (bool, error) {
```
pls add some comments here.
I think it worth mentioning the password change use case?

andyzhangx replied:
done

andyzhangx commented:
@jingxu97 PTAL, added a comment and also added a windows unit test for `pathValide` func, and I found there is not windows unit test running for this repo.

jingxu97 commented:
/approve

ZeroMagic commented:
/lgtm
